rfc/rfc-102/rfc-102.md

```diff
@@ -0,0 +1,231 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements. See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# RFC-102: Vector Search in Apache Hudi
+
+## Proposers
+
+- @rahil-c
+
+## Approvers
+
+- @suryaprasanna
+- @vinothchandar
+- @balaji-varadarajan-ai
+- @yihua
+
+## Abstract
+
+This RFC proposes adding the ability to perform a native **vector similarity search** on Hudi tables.
+
+Building on RFC-100 (unstructured data storage in Hudi), Hudi tables would contain unstructured content (e.g., images, video, documents) as well as the related *embeddings* for those contents.
+
+The next natural requirement for AI/ML workloads on Hudi is to **search these embeddings efficiently**:
+
+- Given a query embedding (e.g., derived from an image, text, or audio snippet),
+- Find the *k* most similar rows in a Hudi table,
+- Return those rows along with a similarity score.
+
+This RFC focuses on:
+
+1. Defining a **user-facing API** (via Hudi Spark SQL integration) for vector search on Hudi tables.
+2. Describing a **baseline implementation** for how we can achieve this in Spark–Hudi integration.
+
+Note this RFC will also discuss at a high level **future enhancements** around a vector index to speed up this search. These details will likely be fleshed out in a future RFC around index creation and management.
+
+## Background
+
+### What is a Vector Embedding?
```

Comment:
add some links for readers to read about this in detail? vs us explaining this here inline..